web/modules/custom/joinup_workflow/joinup_workflow.module

```diff
@@ -277,15 +267,29 @@ function joinup_workflow_is_motivation_required($entity_type_id, $workflow_id, $
  */
 function joinup_workflow_message_form_submit_alter(array &$form, FormStateInterface $form_state) {
   $node = $form_state->getFormObject()->getEntity();
-  if ($node->get('field_state')->getString() == 'draft') {
-    \Drupal::messenger()->deleteByType('status');
-    $args = [
-      '@type' => node_get_type_label($node),
-      '%title' => $node->getTitle(),
-      '%state' => $node->get('field_state')->getString(),
-      ':user' => Url::fromRoute('user.page')->toString(),
-      ':url' => $node->toUrl('canonical')->toString(),
-    ];
-    \Drupal::messenger()->addStatus(t('@type <a href=":url">%title</a> has been created as %state. You can find it in the section "My unpublished content" located in your <a href=":user">My account page</a>, or in the aforementioned section under the Collection/Solution it was created in.', $args));
+
+  if ($node instanceof CommunityContentInterface) {
+    $messenger = \Drupal::messenger();
+    if ($node->get('field_state')->value == 'draft') {
+      $status_messages = $messenger->messagesByType('status');
+      $messenger->deleteByType('status');
+      $args = [
+        '@type' => node_get_type_label($node),
+        '%title' => $node->toLink()->toString(),
+        '%state' => $node->get('field_state')->getString(),
+        ':user' => Url::fromRoute('user.page')->toString(),
+        '@audience' => $node->get('og_audience')->entity->get('rid')->entity->label(),
+      ];
+
+      foreach ($status_messages as $message) {
+        if ($message instanceof MarkupInterface) {
+          if ((string) $message != (string) t('@type %title has been created.', $args)) {
```

Comment:
As we're not using strict comparison, I think we can drop the string casting at least on one side.